api/types.go

```diff
@@ -707,21 +707,72 @@ func (pT PinType) String() string {
 	}
 }
 
+var pinOptionsMetaPrefix = "meta-"
+
 // PinOptions wraps user-defined options for Pins
 type PinOptions struct {
 	ReplicationFactorMin int               `json:"replication_factor_min" codec:"rn,omitempty"`
 	ReplicationFactorMax int               `json:"replication_factor_max" codec:"rx,omitempty"`
 	Name                 string            `json:"name" codec:"n,omitempty"`
 	ShardSize            uint64            `json:"shard_size" codec:"s,omitempty"`
+	UserAllocations      []string          `json:"user_allocations" codec:"ua,omitempty"`
```

Comment:
`[]peer.ID` instead of `[]string`?

Reply:
My idea was that UserAllocations may also in the future provide peer names instead of peer IDs etc. We should be able to declare them as peer.ID since this exists: https://godoc.org/github.com/libp2p/go-libp2p-peer#ID.MarshalJSON but I want to change this in multiple places (so different PR).